src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx

```diff
@@ -79,17 +79,19 @@ const NodeDetailComponent = ({
 
     // States uplifted from Manifest Component
     const manifestViewRef = useRef<ManifestViewRefType>({
-        error: false,
-        secretViewAccess: false,
-        desiredManifest: '',
-        manifest: '',
-        activeManifestEditorData: '',
-        modifiedManifest: '',
-        isEditmode: false,
-        activeTab: '',
+        data: {
+            error: false,
+            secretViewAccess: false,
+            desiredManifest: '',
+            manifest: '',
+            activeManifestEditorData: '',
+            modifiedManifest: '',
+            isEditmode: false,
+            activeTab: '',
+        },
+        key: `${params.nodeType}-${params.podName}`,
```

Comment:
it should contain namespace and other params as well in the key

Reply:
namespace and other params are often undefined. But to be safe we could re-use getComponentKey to have a single source of truth.

@vivek-devtron I was thinking of abstracting getComponentKey into a wrapper with a context. That is create a wrapper that will provide a context with the component key. We will pass key & getKey func to the wrapper and then get the key using the useGetComponentKey or something inside the wrapped Component. Would this be an overkill? it would very rarely be used if at all outside of this one case :)